bluesky_widgets/_matplotlib_axes.py

```diff
@@ -149,10 +149,20 @@ def _add_image(self, image_spec):
         array = image_spec.func(run)
 
         # Initialize artist with currently-available data.
-        artist = self.axes.imshow(array, label=image_spec.label, **image_spec.style, origin='lower')
+        artist = self.axes.imshow(array, label=image_spec.label, origin='lower', **image_spec.style)
         self.axes.set_aspect(self.model.aspect)
-        self.axes.set_xlim(self.model.x_limits)
-        self.axes.set_ylim(self.model.y_limits)
+        # Set x and y limits to defaults if they aren't defined
+        # This is needed for Image to have proper limits
+        if self.model.x_limits is None:
+            self.axes.set_xlim(-0.5, array.shape[1]-0.5)
```

Comment:
This steps outside the paradigm* that all changes slow from the model to the view. Can we make this change entirely on the model side when an image is first added to `AxesSpec`?


*The paradigm is important to avoid confusing race conditions where conflicting actions can drive the application into an inconsistent state. A good example of how this works is, when you close a tab in the view, it tells the model to remove the corresponding element, and the _model_ then tells the view to actually remove that tab. It would be more direct to just close the tab, but by telling the model first, we make sure that the model knows the tab is gone before we actually remove it.